scripts/lib/CIME/XML/env_batch.py

```diff
@@ -108,36 +91,41 @@ def get_type_info(self, vid):
         return type_info
 
     def get_jobs(self):
-        jobs = []
-        for node in self.get_nodes("job"):
-            name = node.get("name")
-            jobs.append(name)
-        return jobs
-
-    def create_job_groups(self, bjobs):
-        # only the job_submission group is repeated
-        group = self.get_node("group", {"id":"job_submission"})
-        # look to see if any jobs are already defined
-        cjobs = self.get_jobs()
-        childnodes = []
+        groups = self.get_nodes("group")
+        results = []
+        for group in groups:
+            if group.get("id") not in ["job_submission", "config_batch"]:
+                results.append(group.get("id"))
+
+        return results
+
+    def create_job_groups(self, batch_jobs):
+        # Subtle: in order to support dynamic batch jobs, we need to remove the
+        # job_submission group and replace with job-based groups
 
-        expect(len(cjobs)==0," Looks like job groups have already been created")
+        orig_group = self.get_optional_node("group", {"id":"job_submission"})
```

Comment:
Important change. job_submission group is removed entirely, each job gets a group now. This allows is to select data by job which we we're doing all over the place, without having to put hacks in for special treatment.